src/main/scala/edu/ie3/powerFactory2psdm/model/powerfactory/PowerFactoryGridMaps.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.powerFactory2psdm.model.powerfactory
+
+import edu.ie3.powerFactory2psdm.exception.pf.{
+  MissingGridElementException,
+  MissingParameterException
+}
+import edu.ie3.powerFactory2psdm.model.powerfactory.PowerFactoryGrid.{
+  Lines,
+  Nodes,
+  Switches
+}
+import java.util.UUID
+import com.typesafe.scalalogging.LazyLogging
+
+class PowerFactoryGridMaps(pfGrid: PowerFactoryGrid) extends LazyLogging {
```

Comment:
The structure of this class does not really serve its purpose.

First and foremost, you use the constructor here for builder pattern purposes. Or with other words: The `PowerFactoryGrid` you hand in is not accessed from the outside, but it's projections onto `node2UUID` etc. In Scala, you would put all those members of the class into the constructor and put all the logic into `PowerFactoryGridMaps#apply` (the companion object).

If you do so, it highlights the difference between data structure (the class) and logic (the companion object) in this example. Later, you only use `PowerFactoryGridMaps` (the class) as a container class for your projections.

Reply:
I like !